com.unity.render-pipelines.core/Tests/Editor/RemoveComponentTests.cs

```diff
@@ -0,0 +1,152 @@
+using NUnit.Framework;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using UnityEditor.Rendering;
+
+namespace UnityEngine.Rendering.Tests
+{
+    class RemoveComponentTests
+    {
+        #region Components
+        class Apple : MonoBehaviour { }
+        class Banana : MonoBehaviour { }
+        [RequireComponent(typeof(Apple))]
+        class AdditionalApple : MonoBehaviour, IAdditionalData { }
+        [RequireComponent(typeof(Banana))]
+        class AdditionalBanana : MonoBehaviour, IAdditionalData { }
+        [RequireComponent(typeof(Banana))]
+        class AdditionalBananaColor : MonoBehaviour, IAdditionalData { }
+        class WaterMelon : MonoBehaviour, IAdditionalData { }
+        [RequireComponent(typeof(Apple), typeof(Banana))]
+        class FruitBasket : MonoBehaviour, IAdditionalData { }
+        #endregion
+
+        class Property
+        {
+            static bool GenericRemoveComponent(
+                [DisallowNull] GameObject gameObject,
+                [DisallowNull] Type componentToRemove,
+                [DisallowNull] Type[] componentsToAdd,
+                [DisallowNull] Type[] expectedComponents,
+                [DisallowNull] Action<Component> removeMethod)
+            {
+                // Add all the components to the game object
+                foreach (var type in componentsToAdd)
+                    gameObject.AddComponent(type);
+
+                // Perform removal
+                removeMethod(gameObject.GetComponent(componentToRemove));
+
+                // The ones that must be removed, are still there?
+                if (!componentsToAdd.Except(expectedComponents).All(type => gameObject.GetComponent(type) == null))
+                    return false;
+
+                // Check expected are there
+                if (!expectedComponents.All(type => gameObject.GetComponent(type) != null))
+                    return false;
+
+                return true;
+            }
+
+            static void RemoveComponent([DisallowNull] Component component) => RemoveComponentUtils.RemoveComponent(component);
+
+            public static bool RemoveComponent(
+                [DisallowNull] GameObject gameObject,
+                [DisallowNull] Type componentToRemove,
+                [DisallowNull] Type[] componentsToAdd,
+                [DisallowNull] Type[] expectedComponents)
+                => GenericRemoveComponent(gameObject, componentToRemove, componentsToAdd, expectedComponents, RemoveComponent);
```

Comment:
Can be simplified with direct call like
```suggestion
                => GenericRemoveComponent(gameObject, componentToRemove, componentsToAdd, expectedComponents, RemoveComponentUtils.RemoveComponent);
```